### DISCUSSION TITLE
setting verified back to false

### DISCUSSION THREAD

**Discussion by MalikWhitten67** [pocketbase/pocketbase#1841] [Q&A]
I am trying to implement a simple two-factor authentication system using the JavaScript SDK of a base mailing verification system. When a user sends a login request, I would like to email them a verification token created by PocketBase. Upon clicking the verify link, the token is sent to my custom URL for validation. The problem I am facing is that every time I click the verify link, the user's verified status is set to true, and I do not know how to set it back to false outside of the admin dashboard. Can you help me with this issue?

**Comment by ganigeorgiev** [comment]
I'm not sure that I understand what do you mean by _"two-factor authentication system using the JavaScript SDK of a base mailing verification system"._

Only admins and users with "Manage API" rule access can change the `verified` field directly.

If you want to implement custom verification, you'll have to [use PocketBase as framework](https://pocketbase.io/docs/use-as-framework/) and attach to the [`OnRecordBeforeConfirmVerificationRequest/OnRecordAfterConfirmVerificationRequest` event hooks](https://pocketbase.io/docs/event-hooks/).

Alternatively, you can also create a separate "customVerified" `bool` field and attach any custom business logic in addition to the default one.

**Reply by MalikWhitten67** [reply]
awww ok

**Reply by MalikWhitten67** [reply]
is there any way to send a custom email then?

**Reply by ganigeorgiev** [reply]
No, there is no web API for that at the moment (_this was requested several times already and I may consider it in the future_).

For sending custom emails programmatically you can check https://pocketbase.io/docs/sending-emails/.

Alternatively, if you don't want to extend PocketBase, you can setup a [nodemailer](https://nodemailer.com/about/) or other similar client (the drawback is that the smtp credentials will be in 2 places).

**Reply by MalikWhitten67** [reply]
ah ok thanks
